extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -73,7 +90,31 @@ function BaseCarouselWithRef(
   const scrollRef = useRef(null);
   const containRef = useRef(null);
   const contentRef = useRef(null);
+  const shouldAutoAdvance = useRef(autoAdvance);
+  const autoAdvanceTimesRef = useRef(0);
+  const autoAdvanceInterval = useMemo(
+    () => Math.max(customInterval, MIN_AUTO_ADVANCE_INTERVAL),
+    [customInterval]
+  );
 
+  const advance = useCallback(
```

Comment:
To clarify: this callback is only used for auto-advance, right? Not for any other types of advance, such as interaction?

Reply:
Yes, would it be helpful to give it a clearer name? For now I've added a comment.

Reply:
Renamed this to `autoAdvance` and renamed the `autoAdvance` prop to `shouldAutoAdvance` ... let me know if this is confusing.